services/web-app/components/catalog/catalog.js

```diff
@@ -153,8 +260,36 @@ function Catalog({ collection, data, type, filter: defaultFilter = {}, glob = {}
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [assets, JSON.stringify(filter), sort, search])
 
+  useEffect(() => {
+    const cleanFilter = Object.fromEntries(
+      Object.entries({ framework, sponsor, platform, tags, status }).filter(([_, v]) => !!v)
+    )
+    if (!isEqual(cleanFilter, filter)) {
+      setFilter(cleanFilter)
+    }
+  }, [framework, sponsor, platform, tags, status, filter])
+
+  useEffect(() => {
+    setPossibleFilterValues(getFilters({ collection, type }))
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [collection, type])
+
+  const updateQueryValues = (newFilter) => {
+    const queryUpdates = {}
+    if (!isEqual(newFilter.sponsor, filter.sponsor)) queryUpdates.sponsor = newFilter.sponsor
+    if (!isEqual(newFilter.platform, filter.platform)) queryUpdates.platform = newFilter.platform
+    if (!isEqual(newFilter.status, filter.status)) queryUpdates.status = newFilter.status
+    if (!isEqual(newFilter.tags, filter.tags)) queryUpdates.tags = newFilter.tags
+
+    if (!isEqual(newFilter.framework, filter.framework)) {
+      queryUpdates.framework = newFilter.framework
+    }
```

Comment:
Went around this one for a while, don't think it's worth it to try to put this into a for loop or something of the sorts because the work that'd need to be done to figure out what `setSponsor()`, `setFramework()`, `setPlatform()` to call would result in way more lines (see new changes)